internal/serve/httphandler/circle_config_handler.go

```diff
@@ -176,10 +176,16 @@ func (h CircleConfigHandler) validateConfigWithCircle(ctx context.Context, patch
 
 	// validate incoming WalletID
 	if patchRequest.WalletID != nil {
-		_, err := circleClient.GetWalletByID(ctx, walletID)
+		accountConfig, err := circleClient.GetAccountConfiguration(ctx)
 		if err != nil {
 			return wrapCircleError(ctx, err)
 		}
+		if accountConfig.Payments == nil || accountConfig.Payments.MasterWalletID == nil {
+			return httperror.BadRequest("Failed to retrieve the master wallet ID from Circle", nil, nil)
+		}
+		if *accountConfig.Payments.MasterWalletID != walletID {
```

Comment:
If you don't use pointers, you can compare `accountConfig.Payments.MasterWalletID != walletID` directly without testing if anything is nil.

Reply:
Addressed in 7e14a61e48b93b4d67935be63c2622ab8d4f8594